src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -206,12 +206,19 @@ private BufferSegment AllocateSegment(int sizeHint)
 
             if (_pool is null)
             {
+                // Use the array pool
                 newSegment.SetMemory(ArrayPool<byte>.Shared.Rent(GetSegmentSize(sizeHint)));
             }
-            else
+            else if (sizeHint <= _pool.MaxBufferSize)
             {
+                // Use the specified pool if it fits
                 newSegment.SetMemory(_pool.Rent(GetSegmentSize(sizeHint, _pool.MaxBufferSize)));
             }
+            else
+            {
+                // We can't use the pool so allocate an array
+                newSegment.SetUnownedMemory(new byte[sizeHint]);
```

Comment:
Should this use ArrayPool as well?

Reply:
It could but I'm not sure how I feel about the user specifying a pool to use but then memory coming from a different pool but only sometimes. So I was thinking that it's easier to explain if it was either your pool or an allocated array.

Reply:
I see.  Ok.